### DISCUSSION TITLE
Getting authentication error when running Apps-of-Apps in a private Github repo

### DISCUSSION THREAD

**Discussion by nthskyradiated** [argoproj/argo-cd#20548] [Q&A]
Hey guys.

I was wondering if somebody could help me figure out what I'm doing wrong. 

Running a simple deployment on a private Github repo works okay using deploy keys. However when I deploy with Apps-of-Apps, I'm getting: _Failed to load target state: failed to generate manifest for source 1 of 1: rpc error: code = Unknown desc = authentication required_ on the children apps. The parent app is syncing ok.

I wonder why that is. I even tried adding my login key in the UI but that doesn't seem help.

Any ideas?

**Comment by nthskyradiated** [comment]
nvm. I figured it out. the repository URL of my children apps is using https which is different from the parent app which uses ssh.

**Reply by prashant0085** [reply]
@nthskyradiated so parent app with ssh url and child with https url was causing the issue and changing url of all child apps to ssh solved it?

I am facing the same and for now I only changed parent app url to ssh for testing.
